vcfa/datasource_vcfa_kube_config.go

```diff
@@ -0,0 +1,157 @@
+				Description: "Bearer token for authentication to the Kubernetes cluster",
+				Sensitive:   true,
+			},
+			"user": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Bearer token username",
+			},
+			"context_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the generated context",
+			},
+			"kube_config_raw": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Raw kubeconfig",
+				Sensitive:   true,
+			},
+		},
+	}
+}
+
+func datasourceVcfaKubeConfigRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	tmClient := meta.(ClientContainer).tmClient
+
+	clusterName := fmt.Sprintf("%s:%s", tmClient.Org, tmClient.Client.VCDHREF.Host)
+	clusterServer := fmt.Sprintf(ccitypes.KubernetesSubpath, tmClient.Client.VCDHREF.Scheme, tmClient.Client.VCDHREF.Host)
+	contextName := tmClient.Org
+
+	projectName, okProjectName := d.GetOk("project_name")
+	supervisorNamespaceName, okSupervisorNamespace := d.GetOk("supervisor_namespace_name")
+	if okProjectName && okSupervisorNamespace {
+		supervisorNamespace, err := readSupervisorNamespace(tmClient, projectName.(string), supervisorNamespaceName.(string))
+		if err != nil {
+			return diag.Errorf("error reading %s: %s", labelSupervisorNamespace, err)
+		}
+		readyStatus := false
+		for _, condition := range supervisorNamespace.Status.Conditions {
+			if strings.ToLower(condition.Type) == "ready" {
+				if strings.ToLower(condition.Status) == "true" {
+					readyStatus = true
+				}
+				break
+			}
+		}
+		if !readyStatus {
+			return diag.Errorf("%s %s is not in a ready status", labelSupervisorNamespace, supervisorNamespaceName)
+		}
+		if supervisorNamespace.Status.NamespaceEndpointURL == "" {
+			return diag.Errorf("unable to retrieve the endpoint URL for %s %s", labelSupervisorNamespace, supervisorNamespaceName)
+		}
+		clusterName = fmt.Sprintf("%s:%s@%s", tmClient.Org, supervisorNamespaceName.(string), tmClient.Client.VCDHREF.Host)
+		clusterServer = supervisorNamespace.Status.NamespaceEndpointURL
+		contextName = fmt.Sprintf("%s:%s:%s", tmClient.Org, supervisorNamespaceName.(string), projectName.(string))
+	}
+
+	token, _, err := new(jwt.Parser).ParseUnverified(tmClient.Client.VCDToken, jwt.MapClaims{})
+	if err != nil {
+		return diag.Errorf("error parsing JWT token: %s", err)
+	}
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return diag.FromErr(errors.New("could not parse claims from JWT token"))
+	}
+	preferredUsername, ok := claims["preferred_username"].(string)
+	if !ok {
+		return diag.FromErr(errors.New("could not parse preferred username from JWT token claims"))
+	}
+	username := fmt.Sprintf("%s:%s@%s", tmClient.Org, preferredUsername, tmClient.Client.VCDHREF.Host)
+
+	kubeconfig := &clientcmdapi.Config{
+		Kind:       "Config",
+		APIVersion: clientcmdapi.SchemeGroupVersion.Version,
+		Clusters: map[string]*clientcmdapi.Cluster{
```

Comment:
This will more likely need to be converted to:
```
Clusters: []clientcmdapi.NamedCluster{
                         Name: clusterName,
			Cluster: Cluster{
				InsecureSkipTLSVerify: tmClient.InsecureFlag,
				Server:                clusterServer,
			},
		},
```

Reply:
Changed.